clash-lib/src/Clash/Primitives/Prelude.hs

```diff
@@ -0,0 +1,44 @@
+{-|
+  Copyright   :  (C) 2023     , QBayLogic B.V.,
+  License     :  BSD2 (see the file LICENSE)
+  Maintainer  :  QBayLogic B.V. <devops@qbaylogic.com>
+
+  Blackbox functions for primitives in one of the @Prelude@ modules.
+-}
+
+module Clash.Primitives.Prelude
+  ( clashCompileErrorBBF
+  ) where
+
+import Control.Monad.State (State)
+import GHC.Stack (withFrozenCallStack)
+import Prelude
+
+import Clash.Backend (Backend)
+import Clash.Netlist.BlackBox.Types (BlackBoxFunction, emptyBlackBoxMeta)
+import Clash.Netlist.BlackBox.Util (exprToString)
+import Clash.Netlist.Types
+  (BlackBox(..), BlackBoxContext(..), TemplateFunction(..))
+import Data.Text.Prettyprint.Doc.Extra (Doc)
+
+clashCompileErrorBBF :: BlackBoxFunction
+clashCompileErrorBBF _isD _primName _args _ty =
+  pure (Right (emptyBlackBoxMeta, bb))
+ where
+  [ _hasCallStack
+    , msg
+    , _ ] = [0..]
+  bb = BBFunction "Clash.Prelude.ExtraBB.clashCompileErrorBBF" 0
```

Comment:
```suggestion
  bb = BBFunction (show 'clashCompileErrorBBF) 0
```

Reply:
Huh. Weird that my eyes really never caught on the string! Thanks for spotting it!